### ISSUE TITLE
Download server too slow (location: germany)

### ISSUE THREAD

**Issue by ghost** [minio/minio#9847]
![Ekran Resmi 2020-06-16 16 15 39](https://user-images.githubusercontent.com/24687234/84779333-f2456c00-afec-11ea-8fe1-224ac3df9ee5.png)

Are you aware of this issue which has been going on for months ?

**Comment by harshavardhana** [comment]
We will check it's hosted ok DigitalOcean we will see why is that the case .

@aead do you see this on your end?

**Comment by klauspost** [comment]
@harshavardhana I've been seeing this as well as @vadmeste - it runs very fast and then suddenly slows down.

**Comment by harshavardhana** [comment]
> @harshavardhana I've been seeing this as well as @vadmeste - it runs very fast and then suddenly slows down.

Hmm not sure why DigitalOcean would be throttling..

**Comment by kotso** [comment]
yes, noticed last couple weeks, when need to download client, speed varies a lot. 

testing from various locations of central and eastern EU.

**Comment by aead** [comment]
Hm interesting. For me it starts quite slow (`~200 KiB/s`) but after a 15/20s or so it becomes quite fast `> 10 MiB/s`. After about 15% it starts speeding up..

**Comment by klauspost** [comment]
Yeah, it is very intermittent. Sometimes starts fast, stuck at 40% for minutes, then suddenly resumes.

**Comment by vadmeste** [comment]
I tested from a dedicated server in France and it works fine.. but seeing the issue while using my local machine. 

`ping -c 30 dl.min.io` can show mdev value.. High mdev value means less quality transferring datay

**Comment by harshavardhana** [comment]
I have blocked an IP which was hogging the network on a continuous basis. Please let us know if the issue persists.

**Comment by jamescorbett** [comment]
I have same issue in the UK. Limiting me to kb/s

**Comment by harshavardhana** [comment]
Exploring if it's possible to move to a different server or a different public IP @jamescorbett

**Comment by brendanbbbb** [comment]
I'm located in Australia and have seen the same very slow downloads (initially when I downloaded it a few months ago). I didn't think much of it at the time until one of my staff commented to me about a week ago about the same issue. Just tested it again and still very slow.

**Comment by jaimem88** [comment]
![Screen Shot 2020-09-28 at 4 21 06 pm](https://user-images.githubusercontent.com/11710811/94397207-ab980e00-01a6-11eb-9753-a98f304733ac.png)

I'm also seeing very slow downloads in Australia 😞

**Comment by torblerone** [comment]
```bash
curl --output minio.RELEASE.2020-10-09T22-55-05Z  http://dl.min.io/server/minio/release/linux-amd64/minio.RELEASE.2020-10-09T22-55-05Z
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 49.4M  100 49.4M    0     0  3375k      0  **0:00:15**  0:00:15 --:--:-- 3826k
```
Download speed from Germany. Not that much time spent

**Comment by JanHolger** [comment]
Download takes 2-3 min for me aswell so i guess this issue still isn't solved.

**Comment by pleclech** [comment]
Downloading from spain also too slow 2min for downloading 19Mb

**Comment by DerDanilo** [comment]
@minio Server admins: Why not just use some servers from Hetzner Cloud, netcup, other providers who don't really throttle and build download replica servers. Just let them sync and have this problem gone. They are fast and cheap. 

As a hint for the other, `axel` is your friend. Building 50+ connections to download one file speeds up things on our end (located in Germany). https://github.com/axel-download-accelerator/axel

**Comment by JanHolger** [comment]
Another workaround might be downloading the docker image tarball from DockerHub and extracting the minio binary. We ended up mirroring it locally instead. The easiest official solution would be uploading artifacts with releases on GitHub. GitHub's azure servers are sometimes terrible aswell but still way better than waiting 15 minutes for a 30mb file.

**Comment by amyspark** [comment]
Ran into this while trying to download minio to a GitLab CI pipeline on France (OVH). I had to use Alpine edge and download their release.

**Comment by tholu** [comment]
```
--2021-05-12 13:09:55--  (try:14)  https://dl.min.io/client/mc/release/linux-amd64/mc
Connecting to dl.min.io (dl.min.io)|178.128.69.202|:443... failed: Connection timed out.
Retrying.

--2021-05-12 13:10:37--  (try:15)  https://dl.min.io/client/mc/release/linux-amd64/mc
Connecting to dl.min.io (dl.min.io)|178.128.69.202|:443... failed: Connection timed out.
Retrying.

--2021-05-12 13:11:18--  (try:16)  https://dl.min.io/client/mc/release/linux-amd64/mc
Connecting to dl.min.io (dl.min.io)|178.128.69.202|:443... failed: Connection timed out.
Retrying.
```

Really frustrating.

**Comment by harshavardhana** [comment]
> ```
> --2021-05-12 13:09:55--  (try:14)  https://dl.min.io/client/mc/release/linux-amd64/mc
> Connecting to dl.min.io (dl.min.io)|178.128.69.202|:443... failed: Connection timed out.
> Retrying.
> 
> --2021-05-12 13:10:37--  (try:15)  https://dl.min.io/client/mc/release/linux-amd64/mc
> Connecting to dl.min.io (dl.min.io)|178.128.69.202|:443... failed: Connection timed out.
> Retrying.
> 
> --2021-05-12 13:11:18--  (try:16)  https://dl.min.io/client/mc/release/linux-amd64/mc
> Connecting to dl.min.io (dl.min.io)|178.128.69.202|:443... failed: Connection timed out.
> Retrying.
> ```
> 
> Really frustrating.

We will be moving to GitHub for downloads from our own infrastructure this is currently a work in progress.

**Comment by ghost** [comment]
> > ```
> > --2021-05-12 13:09:55--  (try:14)  https://dl.min.io/client/mc/release/linux-amd64/mc
> > Connecting to dl.min.io (dl.min.io)|178.128.69.202|:443... failed: Connection timed out.
> > Retrying.
> > 
> > --2021-05-12 13:10:37--  (try:15)  https://dl.min.io/client/mc/release/linux-amd64/mc
> > Connecting to dl.min.io (dl.min.io)|178.128.69.202|:443... failed: Connection timed out.
> > Retrying.
> > 
> > --2021-05-12 13:11:18--  (try:16)  https://dl.min.io/client/mc/release/linux-amd64/mc
> > Connecting to dl.min.io (dl.min.io)|178.128.69.202|:443... failed: Connection timed out.
> > Retrying.
> > ```
> > 
> > 
> >   
> >     
> >     
> > 
> >     
> >     
> > 
> >   
> > 
> > 
> > Really frustrating.
> 
> We will be moving to GitHub for downloads from our own infrastructure this is currently a work in progress.

finally good news !

**Comment by tholu** [comment]
@harshavardhana Any ETA on this change? I have meanwhile rehosted the binary on our own CDN for internal use.

**Comment by harshavardhana** [comment]
> @harshavardhana Any ETA on this change? I have meanwhile rehosted the binary on our own CDN for internal use.

@tholu there is no set ETA yet as it requires some refactoring of our download infrastructure and release management to start using GitHub for our releases.

**Comment by JanHolger** [comment]
Is this a good idea? Since GitHub was bought by Microsoft and at least parts of the infrastructure have been migrated into azure cloud, I started to get problems with downloading files from GitHub and accessing GitHub Pages aswell and i've spoken to several people who had similar experiences. I mean it's probably better than the current servers but even sending out pigeons would be faster. At the moment we are pulling the docker image and extracting the executable so we don't need to wait 30 minutes for a 30 mb download from the official server everytime we update.

**Comment by chotaire** [comment]
`minio                27%[====>               ]  15,45M  11,1KB/s    ETA 11m 54s`

This has been terrible for years. First only from single ISPs, now since a year from all ISPs I am trying. Please get rid of whoever is hosting this.

**Comment by tmoschou** [comment]
Hi,

Located in Australia

This is consistently taking over 5 minutes to download https://dl.min.io/client/mc/release/linux-amd64/mc 

![Screen Shot 2021-06-23 at 11 28 16 am](https://user-images.githubusercontent.com/5567550/123023562-be2e3380-d416-11eb-8ea6-7a35872bb852.png)

**Comment by JanHolger** [comment]
> Hi,
> 
> Located in Australia
> 
> This is consistently taking over 5 minutes to download https://dl.min.io/client/mc/release/linux-amd64/mc
> 
> ![Screen Shot 2021-06-23 at 11 28 16 am](https://user-images.githubusercontent.com/5567550/123023562-be2e3380-d416-11eb-8ea6-7a35872bb852.png)

I can recommend pulling the docker image and extracting the binary (:

**Comment by tmoschou** [comment]
Thanks @JanHolger 

For others interested, if you are building a Docker image with `mc`, Docker has an easy way to do this with [multistage build](https://docs.docker.com/develop/develop-images/multistage-build/#use-an-external-image-as-a-stage)

```Dockerfile
COPY --from=minio/mc /usr/bin/mc /usr/local/bin/mc
```

**Comment by viceice** [comment]
> Thanks @JanHolger
> 
> For others interested, if you are building a Docker image with `mc`, Docker has an easy way to do this with [multistage build](https://docs.docker.com/develop/develop-images/multistage-build/#use-an-external-image-as-a-stage)
> 
> ```dockerfile
> COPY --from=minio/mc /usr/bin/mc /usr/local/bin/mc
> ```

Doesn't work for windows 😕

**Comment by JanHolger** [comment]
> > Thanks @JanHolger
> > For others interested, if you are building a Docker image with `mc`, Docker has an easy way to do this with [multistage build](https://docs.docker.com/develop/develop-images/multistage-build/#use-an-external-image-as-a-stage)
> > ```dockerfile
> > COPY --from=minio/mc /usr/bin/mc /usr/local/bin/mc
> > ```
> 
> Doesn't work for windows

Quick fix is to uninstall Windows and install a proper os (;

**Comment by viceice** [comment]
Haha, tell your customer to install Linux because minio download servers are too slow. 🤦‍♂️

**Comment by chotaire** [comment]
> Quick fix is to uninstall Windows and install a proper os (;

No, the quick fix is to finally get rid of slow download servers after only 14 months of complaints.

**Comment by JanHolger** [comment]
> > Quick fix is to uninstall Windows and install a proper os (;
> 
> No, the quick fix is to finally get rid of slow download servers after only 14 months of complaints.

Totally agree but under *nix systems we at least have a workaround (: @harshavardhana announced that they switch to a different hosting provider / github downloads in May but nothing really changed since then. I thought it was a german problem only but in the meanwhile I had contact to people from other eastern-european countries who experienced similar issues.

**Comment by klauspost** [comment]
Locking this topic since there it seems to have drifted off topic. 

TLDR; We know it is slow - we are working on an alternative.

**Comment by harshavardhana** [comment]
> TLDR; We know it is slow - we are working on an alternative.

After a lot of understanding looks like DO is very poor and their US routes are fully saturated. Downloading within DO

SFO2 -> NYC3 

```
--2021-11-26 20:20:16--  https://dl.min.io/server/minio/linux-arm6vl.tar.gz
Reusing existing connection to dl.min.io:443.
HTTP request sent, awaiting response... 200 OK
Length: 48461328 (46M) [application/gzip]
Saving to: ‘dl.min.io/server/minio/linux-arm6vl.tar.gz’

dl.min.io/server/minio/linux-arm6vl.t   5%[===>                                                                      ]   2.74M  19.3KB/s    eta 15m 34s
```

We are planning to move to a different provider and this may take some time.

**Comment by harshavardhana** [comment]
All dl.min.io for community releases from the past has been moved to github.com so this is finally resolved.
